plugins/inputs/smart/smart.go

```diff
@@ -1045,7 +1047,17 @@ func parsePercentageInt(fields, deviceFields map[string]interface{}, str string)
 }
 
 func parseDataUnits(fields, deviceFields map[string]interface{}, str string) error {
-	units := strings.Fields(str)[0]
+	// Remove everything after '[' and stip non-numeric chars
+	data := strings.Split(str, "[")
+
+	regex, err := regexp.Compile("[^0-9]+")
+	if err != nil {
+		return fmt.Errorf("failed to compile numeric regex")
+	}
+
+	// '16,626,888 [8,51 TB]' --> 16626888
+	// '16 829 004 [8,61 TB]' --> 16829004
+	units := regex.ReplaceAllString(data[0], "")
```

Comment:
It would work for every case if you move this to `parseCommaSeparatedInt`. 

And maybe change name of `parseCommaSeparatedInt` to something more general? :)

Reply:
@zak-pawel, I have pushed another set of commits. This will remove non-utf8 characters before going through and removing the non-numeric values. Can you give that a shot once the artifacts are attached?

Reply:
@powersj I've checked Polish, Italian (Switzerland) and English locales. I can confirm that all values are properly parsed for every locale mentioned.

```
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Critical_Warning,serial_no=xxx raw_value=0i 1650049271000000000
smart_attribute,device=sdd,host=zak,id=194,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Temperature_Celsius,serial_no=xxx raw_value=54i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Available_Spare,serial_no=xxx raw_value=100i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Available_Spare_Threshold,serial_no=xxx raw_value=10i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Percentage_Used,serial_no=xxx raw_value=0i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Data_Units_Read,serial_no=xxx raw_value=17098088i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Data_Units_Written,serial_no=xxx raw_value=17536456i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Host_Read_Commands,serial_no=xxx raw_value=212453759i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Host_Write_Commands,serial_no=xxx raw_value=241808679i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Controller_Busy_Time,serial_no=xxx raw_value=718i 1650049271000000000
smart_attribute,device=sdd,host=zak,id=12,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Power_Cycle_Count,serial_no=xxx raw_value=809i 1650049271000000000
smart_attribute,device=sdd,host=zak,id=9,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Power_On_Hours,serial_no=xxx raw_value=1339i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Unsafe_Shutdowns,serial_no=xxx raw_value=10i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Media_and_Data_Integrity_Errors,serial_no=xxx raw_value=0i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Error_Information_Log_Entries,serial_no=xxx raw_value=1020i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Warning_Temperature_Time,serial_no=xxx raw_value=0i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Critical_Temperature_Time,serial_no=xxx raw_value=0i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Temperature_Sensor_1,serial_no=xxx raw_value=54i 1650049271000000000
smart_attribute,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,name=Temperature_Sensor_2,serial_no=xxx raw_value=80i 1650049271000000000
smart_device,device=sdd,host=zak,model=Samsung\ SSD\ 970\ EVO\ 1TB,serial_no=xxx temp_c=54i,exit_status=0i,health_ok=true 1650049271000000000
```

Reply:
thank you!